components/hashglobe/src/diagnostic.rs

```diff
@@ -54,17 +60,19 @@ impl<K: Hash + Eq, V, S: BuildHasher> DiagnosticHashMap<K, V, S>
 
     fn verify(&self) {
         let mut position = 0;
+        let mut count = 0;
         let mut bad_canary: Option<(usize, *const usize)> = None;
         for (_,v) in self.map.iter() {
             let canary_ref = &v.0;
+            position += 1;
             if *canary_ref == CANARY {
-                position += 1;
                 continue;
             }
+            count += 1;
             bad_canary = Some((*canary_ref, canary_ref));
```

Comment:
we should keep track of the position of the bad_canary within this Option and pass that down to report_corruption instead